packages/next/src/shared/lib/app-router-context.shared-runtime.ts

```diff
@@ -96,7 +98,9 @@ export type ReadyCacheNode = {
    */
   lazyData: null
   prefetchHead: React.ReactNode
-  head: React.ReactNode
+
+  /** metadata cache nodes, [viewport, metadata] */
+  head: [React.ReactNode, React.ReactNode]
```

Comment:
maybe worth extracting into a shared type for all the spots where we re-create the type? Would potentially have the added benefit of allowing us to comment on what the significance of the tuple ordering is in a single place

Reply:
Created a new `HeadData` type for this case. `React.ReactNode` also contains `null` hence I replaced the `[ReactNode | null, ReactNode | null]` case with this as well